### ISSUE TITLE
`/ignore ` command is broken

### ISSUE THREAD

**Issue by bmorel** [Unvanquished/Unvanquished#2112]
There is a `/ignore ` command to ignore a player. This command is meant to hide private messages and chat messages of that player.

Currently, the command does not prevents the bell sound to be played, and only works randomly, as in, it might or not work in the course of a whole game, without any clue about why it would stop to work.

**Comment by bmorel** [comment]
So, after further analysis of the code, the `struct clientList_t` from daemon is used. This struct is used only in gamelogic and thus does not belongs to engine.
This struct is really a bitfield of length 64 bits, which effectively prevents MAX_CLIENTS to be higher than 64.

There is also no saving whatsoever of that list of ignored people, so on every new instance, new /ignore commands have to be made, "in the hope" that the ignore person is still around.

This structure is also used to remember which players are "ready" to go on next map. In this case it's not much of a problem to not allow persistence.
There is another use about vote restrictions, but the underlying syscall is not implemented, so this does  not work. It seems it didn't worked in some trem implementations neither.

The most basic things to do would be to move this completely to gamelogic (instead of engine) and to replace the bitfield with a `std::bitset<MAX_CLIENTS>`, but this would not fix `/ignore` at all, it would only make it safer if someone decides to compile engine for more clients.

I think the GUID of the ignored person should be added to a new entry in `admin.dat`, and when a player connects or disconnects, the server should check for this list and updates players info accordingly.

**Comment by DolceTriade** [comment]
Yeah... the implementation is very naive... `/ignore` is a client side command right? I'm not sure we should make this a server side thing. I think the client side should be made smarter, even if the server has to send a stable unique identifier for each client.

**Comment by bmorel** [comment]
> even if the server has to send a stable unique identifier for each client.

This is also my opinion, but I believe @slipher disagrees. I do not know his reasons though.

A client-side implementation would allow the ignore list to survive across servers being the most notable advantage I can see.

As for current implementation, I do not yet know all the ins and outs, but it seems to be mostly server-side, with some stuff sent to requester client.

**Comment by slipher** [comment]
My proposal is a server-side file (probably a new one rather than admin.dat) with pairs of GUIDs X ignores Y. This allows ignoring to persist across connections without compromising privacy by sending players' GUIDs. (I *think* GUIDs are supposed to be something of a secret; there were Tremulous QVMs that would let random people read everyone's GUID with listplayers but I don't think we have anything like that.)

**Comment by bmorel** [comment]
Are not GUID some kind of fingerprint of the public key? But I guess you're actually right, the privacy argument is strong, especially considering things like GDPR.

**Comment by DolceTriade** [comment]
In trem, guids were (badly) kept secret because there was no verification done on the guid. The guid was the md5sum of the qkey. If you stole someone's guid, you could steal their admin on another server. (Trem introduced the guid hashed on the server name as workaround).

Unv has pubkey authentication, so stealing someone's GUID is not a valid vector of attack (as the guid is meaningless... in Unv, the server is presented with the client's public key and the server hashes that to produce a guid)

**Comment by DolceTriade** [comment]
Also note that sending the client a stable identifier doesn't have to be the guid (though in practice it would probably be some value derived from the guid).

Also we should try to ensure that the /ignore command extends to voip (which is basically unused, but support still exists).

**Comment by bmorel** [comment]
It seems `ignore` simply does not work at all as long as you're not in a team.
